### PR TITLE
burst_when_thrown_filled item property and condom to use it

### PR DESCRIPTION
**Pull request by Kaltag1925**
#### Summary

SUMMARY: Content "Condom that bursts when thrown filled"

#### Purpose of change

It kind of start off as a joke in the HWS discord channel, but I thought it would be cool to add this mechanic. Basically if an item is thrown that has the property "burst_when_thrown_filled", the code checks what percentage the container is filled and compares it to the number for the property.

#### Describe the solution

Like I said, it checks if the item is a container and has the property then checks if it is filled more than or equal to the percentage given by the property. If it is, the item bursts, gets destroyed, and spills its contents. It's pretty similar to the shatter effect, which is what I based it off of.

#### Describe alternatives you've considered

I was originally doing to do a flag similar to "LIGHT_X" or "RECOVER_X" but I couldn't find a way to get a list of the flags the item had, and it seemed more fitting for it to be a property.

#### Additional context

I don't really think it needs a screen shot, it's kind of simple and self-explanatory, but I can give one if needed.

### CODE REVIEW

File: SIGNOFF [hunk 1]
```diff
@@ -61,3 +61,4 @@ Patrik Stridvall <patrik@stridvall.se>
 Ryan Schneider (RadHazard) <rslego@gmail.com>
 Paul Bransford (draeath) <draeath@gmail.com>
 Mark Langsdorf (mlangsdorf) <mark.langsdorf@gmail.com> / <mlangsdo@redhat.com>
+Davis Bray (Kaltag1925) <pdavisbray@gmail.com>
```
Comment (ZhilkinSerg) [inline SIGNOFF]:
```suggestion
Davis Bray (Kaltag1925) <pdavisbray@gmail.com>

```

File: msvc-full-features/Cataclysm-vcpkg-static.vcxproj [hunk 2]
```diff
@@ -23,7 +23,7 @@
     <ProjectGuid>{19F0BE17-3DAF-40E8-A9D2-904A56382E54}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
     <RootNamespace>Cataclysm</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.17134.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0.17763.0</WindowsTargetPlatformVersion>
```
Comment (ZhilkinSerg) [inline msvc-full-features/Cataclysm-vcpkg-static.vcxproj]:
```suggestion
    <WindowsTargetPlatformVersion>10.0.17134.0</WindowsTargetPlatformVersion>
```

File: msvc-full-features/Cataclysm-vcpkg-static.vcxproj [hunk 3]
```diff
@@ -242,4 +242,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
+</Project>
```
Comment (ZhilkinSerg) [inline msvc-full-features/Cataclysm-vcpkg-static.vcxproj]:
```suggestion
</Project>

```

**Comment by Kaltag1925** [comment]
I was thinking of adding a sound for bursting, but I didn't know what it would be (I mean obviously its rubber), but I don't know if other materials could burst aswell, Should I do some check to see if its a rubber/plastic object popping?

**Comment by anothersimulacrum** [comment]
The condom item already exists, in data/json/items/generic.

**Comment by Kaltag1925** [comment]
Would it be a good idea to add something like "it feels like it's going to burst" to notify the player?

File: msvc-full-features/Cataclysm-vcpkg-static.vcxproj (hunk 3 above)
Comment (Kaltag1925) [reply msvc-full-features/Cataclysm-vcpkg-static.vcxproj]: I messed up and I am unable to accept it, but I made the changes myself so it should be good.

**Comment by Kaltag1925** [comment]
Hey, I'm just curious if there is anything that needs to be done before it gets merged? Thanks

**Pull request merged**